exchanges/deribit/deribit_websocket_eps.go

```diff
@@ -0,0 +1,2139 @@
+package deribit
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/order"
+)
+
+// WSRetriveBookBySummary retrives book summary data for currency requested through websocket connection.
+func (d *Deribit) WSRetriveBookBySummary(currency, kind string) ([]BookSummaryData, error) {
+	if currency == "" {
+		return nil, fmt.Errorf("%w \"%s\"", errInvalidCurrency, currency)
+	}
+	input := &struct {
+		Currency string `json:"currency"`
+		Kind     string `json:"kind,omitempty"`
+	}{
+		Currency: currency,
+	}
+	if kind != "" {
+		input.Kind = kind
+	}
+	var resp []BookSummaryData
+	return resp, d.SendWSRequest(getBookByCurrency, input, &resp, false)
+}
+
+// WSRetriveBookSummaryByInstrument retrives book summary data for instrument requested through the websocket connection.
+func (d *Deribit) WSRetriveBookSummaryByInstrument(instrument string) ([]BookSummaryData, error) {
+	if instrument == "" {
+		return nil, fmt.Errorf("%w, instrument_name is missing", errInvalidInstrumentName)
+	}
+	input := &struct {
+		Instrument string `json:"instrument_name,omitempty"`
+	}{
+		Instrument: instrument,
+	}
+	var resp []BookSummaryData
+	return resp, d.SendWSRequest(getBookByInstrument, input, &resp, false)
+}
+
+// WSRetriveContractSize retrives contract size for instrument requested through the websocket connection.
+func (d *Deribit) WSRetriveContractSize(instrument string) (*ContractSizeData, error) {
+	if instrument == "" {
+		return nil, fmt.Errorf("%w, instrument_name is missing", errInvalidInstrumentName)
+	}
+	input := &struct {
+		Instrument string `json:"instrument_name"`
+	}{
+		Instrument: instrument,
+	}
+	var resp ContractSizeData
```

Comment:
var resp ContractSizeData -> var resp *ContractSizeData

Reply:
This thing returns a nil pointer exception

`panic: runtime error: invalid memory address or nil pointer dereference [recovered]`